target/linux/bcm5862x/image/Makefile

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright (C) 2021 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+include $(TOPDIR)/rules.mk
+include $(INCLUDE_DIR)/image.mk
+
+define Build/meraki-hdr
+	SIZE=$$(printf "%08X" $$(stat -c "%s" $@)) ; \
+	MAGIC=$(MAGIC) ; if [ -z "$$MAGIC" ] ; then MAGIC="\\x4d\\x49\\x50\\x53" ; fi ; \
+	LDADDR=$$(echo $(KERNEL_LOADADDR) | cut -dx -f2) ; if [ -z "$$LDADDR" ] ; then \
+	LDADDR="60008000" ; fi ; \
+	ENTRY=$$(echo $(ENTRY_POINT) | cut -dx -f2) ; if [ -z $$ENTRY ] ; then \
+	ENTRY=$$LDADDR ; fi ; \
+	( echo -ne $$MAGIC | od -tx4 -N4 -An | xxd -r -p | \
+		dd bs=4 count=1 conv=sync ; \
+	echo -ne "\x$${LDADDR:6:2}\x$${LDADDR:4:2}\x$${LDADDR:2:2}\x$${LDADDR:0:2}" | \
+		dd bs=4 count=1 conv=sync ; \
+	echo -ne "\x$${SIZE:6:2}\x$${SIZE:4:2}\x$${SIZE:2:2}\x$${SIZE:0:2}" | \
+		dd bs=4 count=1 conv=sync ; \
+	echo -ne "\x$${ENTRY:6:2}\x$${ENTRY:4:2}\x$${ENTRY:2:2}\x$${ENTRY:0:2}" | \
+		dd bs=4 count=1 conv=sync ; \
+		dd if=/dev/zero bs=1 count=16 conv=sync \
+	) | cat - $@ > $@.new
+	CRC32=$$(crc32 $@.new) && \
+	echo -ne "\x$${CRC32:6:2}\x$${CRC32:4:2}\x$${CRC32:2:2}\x$${CRC32:0:2}" | \
+		dd conv=notrunc of=$@.new seek=16 bs=1 count=4 && \
+	mv $@.new $@
+endef
+
+define Device/meraki_mx64
+	DEVICE_VENDOR := Cisco Meraki
+	DEVICE_MODEL := MX64
+	DEVICE_DTS := bcm958625-mx64
+	DEVICE_PACKAGES := kmod-eeprom-at24
+	BLOCKSIZE := 128k
+	PAGESIZE := 2048
+	KERNEL_SIZE := 3145728
+	KERNEL_NAME := zImage
+	KERNEL = kernel-bin | append-dtb | pad-to $$(BLOCKSIZE) | meraki-hdr
+	UBINIZE_OPTS := -E 5
+	BOARD_NAME := meraki,mx64
+	SUPPORTED_DEVICES += meraki,mx64
+	IMAGES = sysupgrade.bin
+	IMAGE/sysupgrade.bin := append-kernel | sysupgrade-tar kernel=$$$$@ | append-metadata
+endef
+TARGET_DEVICES += meraki_mx64
+
+define Device/meraki_mx64a0
+	DEVICE_VENDOR := Cisco Meraki
+	DEVICE_MODEL := MX64(A0)
+	DEVICE_DTS := bcm958625-mx64a0
+	DEVICE_PACKAGES := kmod-eeprom-at24
+	BLOCKSIZE := 128k
+	PAGESIZE := 2048
+	KERNEL_SIZE := 3145728
+	KERNEL_NAME := zImage
+	KERNEL = kernel-bin | append-dtb | pad-to $$(BLOCKSIZE) | meraki-hdr
+	UBINIZE_OPTS := -E 5
+	BOARD_NAME := meraki,mx64
+	SUPPORTED_DEVICES += meraki,mx64
+	IMAGES = sysupgrade.bin
+	IMAGE/sysupgrade.bin := append-kernel | sysupgrade-tar kernel=$$$$@ | append-metadata
```

Comment:
Please add a Device/Default block like on other targets.
SUPPORTED_DEVICES should be set by an algorithm.